src/test/coinscachepair_tests.cpp

```diff
@@ -139,56 +139,55 @@ BOOST_AUTO_TEST_CASE(linked_list_random_deletion)
     BOOST_CHECK_EQUAL(sentinel.second.Prev(), &sentinel);
 }
 
-BOOST_AUTO_TEST_CASE(linked_list_add_flags)
+BOOST_AUTO_TEST_CASE(linked_list_set_state)
 {
     CoinsCachePair sentinel;
     sentinel.second.SelfRef(sentinel);
     CoinsCachePair n1;
     CoinsCachePair n2;
 
-    // Check that adding DIRTY flag inserts it into linked list and sets flags
+    // Check that setting DIRTY inserts it into linked list and sets state
     CCoinsCacheEntry::SetDirty(n1, sentinel);
-    BOOST_CHECK_EQUAL(n1.second.GetFlags(), CCoinsCacheEntry::DIRTY);
+    BOOST_CHECK(n1.second.IsDirty() && !n1.second.IsFresh());
     BOOST_CHECK_EQUAL(n1.second.Next(), &sentinel);
     BOOST_CHECK_EQUAL(n1.second.Prev(), &sentinel);
     BOOST_CHECK_EQUAL(sentinel.second.Next(), &n1);
     BOOST_CHECK_EQUAL(sentinel.second.Prev(), &n1);
 
-    // Check that adding FRESH flag on new node inserts it after n1
+    // Check that setting FRESH on new node inserts it after n1
     CCoinsCacheEntry::SetFresh(n2, sentinel);
-    BOOST_CHECK_EQUAL(n2.second.GetFlags(), CCoinsCacheEntry::FRESH);
+    BOOST_CHECK(n2.second.IsFresh() && !n2.second.IsDirty());
     BOOST_CHECK_EQUAL(n2.second.Next(), &sentinel);
     BOOST_CHECK_EQUAL(n2.second.Prev(), &n1);
     BOOST_CHECK_EQUAL(n1.second.Next(), &n2);
     BOOST_CHECK_EQUAL(sentinel.second.Prev(), &n2);
 
-    // Check that we can add extra flags, but they don't change our position
+    // Check that we can set extra state, but they don't change our position
     CCoinsCacheEntry::SetFresh(n1, sentinel);
-    BOOST_CHECK_EQUAL(n1.second.GetFlags(), CCoinsCacheEntry::DIRTY | CCoinsCacheEntry::FRESH);
+    BOOST_CHECK(n1.second.IsDirty() && n1.second.IsFresh());
     BOOST_CHECK_EQUAL(n1.second.Next(), &n2);
     BOOST_CHECK_EQUAL(n1.second.Prev(), &sentinel);
     BOOST_CHECK_EQUAL(sentinel.second.Next(), &n1);
     BOOST_CHECK_EQUAL(n2.second.Prev(), &n1);
 
-    // Check that we can clear flags then re-add them
+    // Check that we can clear state then re-set them
```

Comment:
nit: `// Check that we can clear state then re-set it`

Reply:
Done